### PR TITLE
Update `postgresql` functions to use util `expandReference`

### PR DESCRIPTION
**Pull request by mtuchi**
## Summary

Update all functions to use util `exapandReferences` and add `composeNextState` in `findValue` function

- Ref #604 

## Review Checklist

Before merging, the reviewer should check the following items:

- [x] Does the PR do what it claims to do?
- [ ] Are there any unit tests? Should there be?
- [x] Is there a changeset associated with this PR? Should there be? Note that
      dev only changes don't need a changeset.

### CODE REVIEW

File: packages/postgresql/src/Adaptor.js [hunk 1]
```diff
@@ -157,23 +157,25 @@ function queryHandler(state, query, options, callback) {
...
- * string of SQL.
+ * @param {string} sqlString - A sequence of characters that forms a valid SQL command.
  * @param {object} [options] - Optional options argument
  * @param {boolean} [options.writeSql] - A boolean value that specifies whether to log the generated SQL statement. Defaults to false.
  * @param {boolean} [options.execute] - A boolean value that specifies whether to execute the generated SQL statement. Defaults to false.
  * @param {function} callback - (Optional) callback function
  * @returns {Operation}
  */
-export function sql(sqlQuery, options, callback) {
+export function sql(sqlString, options, callback) {
```
Comment (josephjclark) [inline packages/postgresql/src/Adaptor.js:167]: I think `sqlQuery` is still appropriate? It's a string or a function that returns a string.

I agree that the old way was too dogmatic and that making it an expandable string is correct. I just don't think we should rename it.

Also `sequence of characters that forms a valid SQL command` is strange phrasing. I'd just say "the SQL query as a string"

File: packages/postgresql/src/Adaptor.js [hunk 2]
```diff
@@ -234,12 +240,16 @@ export function findValue(filter) {
...
+              ...composeNextState(state, returnValue),
+              result: returnValue,
+            };
+            resolve(nextState);
           }
         });
       });
     } catch (e) {
-      client.end();
+      console.log(e, 'asd');
```
Comment (josephjclark) [inline packages/postgresql/src/Adaptor.js:252]: This logging needs removing. And are you sure you don't want to end the client here?

Comment (josephjclark) [reply packages/postgresql/src/Adaptor.js:252]: Wait does this even fire?

Ach, I don't want to touch code I don't understand if I don't need to..

File: .changeset/green-monkeys-compete.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+'@openfn/language-postgresql': minor
+---
+
+- Update all operations to use util `expandReferences`
+- Add `findValue` result to state
```
Comment (josephjclark) [inline .changeset/green-monkeys-compete.md:6]: This is a major change

File: packages/postgresql/src/Adaptor.js [hunk 4]
```diff
@@ -264,23 +274,31 @@ export function findValue(filter) {
...
+        ),
+        handleOptions(resolvedOptions)
       );
 
-      const safeQuery = `INSERT INTO ${table} (${columnsList}) VALUES [--REDACTED--]];`;
+      const safeQuery = `INSERT INTO ${resolvedTable} (${columnsList}) VALUES [--REDACTED--]];`;
 
-      const queryToLog = options && options.logValues ? query : safeQuery;
+      const queryToLog =
+        resolvedOptions && resolvedOptions.logValues ? query : safeQuery;
```
Comment (josephjclark) [inline packages/postgresql/src/Adaptor.js:299]: Try `resolvedOptions?.logValues ? query : safeQuery;`

File: packages/postgresql/src/Adaptor.js [hunk 5]
```diff
@@ -307,30 +325,37 @@ export function insert(table, record, options, callback) {
...
+          ),
+          handleOptions(resolvedOptions)
         );
 
-        const safeQuery = `INSERT INTO ${table} (${columnsList}) VALUES [--REDACTED--]];`;
+        const safeQuery = `INSERT INTO ${resolvedTable} (${columnsList}) VALUES [--REDACTED--]];`;
 
-        const queryToLog = options && options.logValues ? query : safeQuery;
+        const queryToLog =
+          resolvedOptions && resolvedOptions.logValues ? query : safeQuery;
```
Comment (josephjclark) [inline packages/postgresql/src/Adaptor.js:356]: Again, as we're touching it anyway, `resolvedOptions.logValues ? query : safeQuery;`

File: packages/postgresql/src/Adaptor.js [hunk 6]
```diff
@@ -363,14 +388,16 @@ export function insertMany(table, records, options, callback) {
  */
 export function upsert(table, uuid, record, options, callback) {
   return state => {
-    let { client } = state;
+    const { client } = state;
 
+    const [resovledTable, resovledUuid, resolvedRecord, resolvedOptions] =
```
Comment (josephjclark) [inline packages/postgresql/src/Adaptor.js:393]: Typos aplenty here, eg `resovledUuid`

**Review by josephjclark — CHANGES_REQUESTED** [review]
Thanks @mtuchi , just a couple of things

File: packages/postgresql/src/Adaptor.js (hunk 1 above)
Comment (mtuchi) [reply packages/postgresql/src/Adaptor.js:167]: Well you can pretty much write an sql command Eg: `create table tableName (column1 dataType)`,  To me `sqlQuery` means only write sql queries. Eg: `select * from patients`

File: packages/postgresql/src/Adaptor.js (hunk 2 above)
Comment (mtuchi) [reply packages/postgresql/src/Adaptor.js]: Removed the `console.log()`

Comment (josephjclark) [reply packages/postgresql/src/Adaptor.js]: No worries, I do that sort of thing all the time :) This is why we self-review

**Review by josephjclark — APPROVED** [review]

**Pull request merged**